core/services/workflows/engine.go

```diff
@@ -21,98 +24,126 @@ const (
 
 type Engine struct {
 	services.StateMachine
-	logger     logger.Logger
-	registry   types.CapabilitiesRegistry
-	trigger    capabilities.TriggerCapability
-	consensus  capabilities.ConsensusCapability
-	targets    []capabilities.TargetCapability
-	workflow   *Workflow
-	callbackCh chan capabilities.CapabilityResponse
-	cancel     func()
+	logger       logger.Logger
+	registry     types.CapabilitiesRegistry
+	workflow     *workflow
+	store        *store
+	queue        *queue[stepRequest]
+	callbackCh   chan capabilities.CapabilityResponse
+	newWorkerCh  chan struct{}
+	stepUpdateCh chan stepState
+	wg           *sync.WaitGroup
```

Comment:
this looks odd. i think wg are usually not pointers.

Reply:
Yeah agreed, I think I made it a pointer so that I could share it with `queue.start` -- just passing a value led to a linting error (copylock). Usually I think a wg would be captured via a closure, therefore bypassing this problem, but not sure how to do that in the context of the queue struct.

Reply:
You can still share a value field as a pointer by just taking the address `&e.wg`

Reply:
I'm aware, my thinking was to just store it as a pointer, but thinking back on it I agree that doesn't make sense.